src/main/java/com/blamejared/compat/tcomplement/highoven/HighOven.java

```diff
@@ -0,0 +1,284 @@
+		init();
+		ModTweaker.LATE_ADDITIONS
+				.add(new HighOven.AddHeat(InputHelper.toFluid(output), InputHelper.toFluid(input), temp + 300));
+	}
+
+	private static class AddHeat extends BaseAction {
+		private FluidStack output, input;
+		private int temp;
+
+		public AddHeat(FluidStack output, FluidStack input, int temp) {
+			super("HighOven.HeatRecipe");
+			this.output = output;
+			this.input = input;
+			this.temp = temp;
+		}
+
+		@Override
+		public void apply() {
+			TCompRegistry.registerHeatRecipe(new HeatRecipeTweaker(input, output, temp));
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Adding %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		protected String getRecipeInfo() {
+			return LogHelper.getStackDescription(output);
+		}
+	}
+
+	private static class RemoveHeat extends BaseAction {
+		private ILiquidStack input;
+		private ILiquidStack output;
+
+		public RemoveHeat(ILiquidStack input, ILiquidStack output) {
+			super("HighOven.HeatRecipe");
+			this.input = input;
+			this.output = output;
+		}
+
+		@Override
+		public void apply() {
+			REMOVED_HEAT_RECIPES.put(output, input);
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Removing %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		public String getRecipeInfo() {
+			return LogHelper.getStackDescription(output);
+		}
+
+	}
+
+	@ZenMethod
+	public static void removeMixRecipe(ILiquidStack output) {
+		init();
+		CraftTweakerAPI.apply(new RemoveMix(output));
+	}
+
+	@ZenMethod
+	public static MixRecipeHelper newMixRecipe(ILiquidStack output, ILiquidStack input, int temp) {
+		init();
+		return new MixRecipeHelper(output, input, temp);
+	}
+
+	private static class RemoveMix extends BaseAction {
+		private ILiquidStack output;
+
+		public RemoveMix(ILiquidStack output) {
+			super("HighOven.MixRecipe");
+			this.output = output;
+		}
+
+		@Override
+		public void apply() {
+			REMOVED_MIX_RECIPES.add(output);
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Removing %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		protected String getRecipeInfo() {
+			return LogHelper.getStackDescription(output);
+		}
+	}
+
+	@SubscribeEvent
+	public void onTinkerRegister(TCompRegisterEvent.HighOvenFuelRegisterEvent event) {
+		if (event.getRecipe() instanceof HighOvenFuelTweaker) {
+			return;
+		}
+		for (IItemStack entry : REMOVED_FUELS) {
+			if (event.getRecipe().matches(InputHelper.toStack(entry))) {
+				event.setCanceled(true);
+			}
+		}
+	}
+
+	@SubscribeEvent
+	public void onTinkerRegister(TCompRegisterEvent.HighOvenHeatRegisterEvent event) {
+		if (event.getRecipe() instanceof HeatRecipeTweaker) {
+			return;
+		} else {
+			for (Map.Entry<ILiquidStack, ILiquidStack> entry : REMOVED_HEAT_RECIPES.entrySet()) {
+				if (event.getRecipe().getOutput().isFluidEqual((FluidStack) entry.getKey().getInternal())) {
```

Comment:
Just use `event.getRecipe().matches(FluidStack,FluidStack)`, I coded it originally for this purpose. The method takes two fluid stack parameters treating null as a wildcard for each, and can be overridden if someone adds a custom high oven recipe type (so your code will still match properly)

Reply:
Oh, that's easier and safer, i.e. definitely better - I'll use that.

I realized their was a problem with removing heat/mix recipes for a given output, but for multiple inputs (e.g. remove **iron -> steel** and **lava -> steel** but not **aluminium -> steel**). Due to the `Map<>` structure, only the last one would be applied.
I switched to a `Map<K, Set<V> >` structure but I'm unsure how to handle the recipe comparison. I see two options:
+ The keys are `K instanceof FluidStack`, conversion from `ILiquidStack` happens in the `remove` method. When cancelling events, I `get(event.getRecipe().getOutput())` the `Set<FluidStack>` of forbidden inputs from the `Map`, and then check if `event.getRecipe().getInput()` is in the set. This prevents iterating on all removed entries for each event, **but** I'm not sure the ILiquidStack comparison will be correct and handle all cases
+ (current) I do not use the `get` method and blindly iterate on all entries in the `Map<ILiquidStack, Set<ILiquidStack> >`, checking each time if `event.getRecipe().matches()` holds. I'm sure the comparison will be correct, but the iteration is sub-optimal.

In both case, a mapping from `K` to `null` indicate the method with null input as wildcard was used, and I cancel all event with the given output.

Should I switch to map/set lookup ? The performance gain should be small and I'm not sure it'll work properly, so it might not be worth the change.